pkg/adapter/adapter_test.go

```diff
@@ -0,0 +1,688 @@
+			textPbPath: `
+								elem: <name: "components" >
+								elem: <
+									name: "component"
+									key: <key: "foo" value: "swpri1-1-1" >
+								>
+								elem: <name: "bar" >`,
+			wantRetCode: codes.NotFound,
+		}, {
+			desc:        "use of model data not supported",
+			modelData:   []*pb.ModelData{{}},
+			wantRetCode: codes.Unimplemented,
+		}, {
+			desc: "device fails to get",
+			textPbPath: `
+			elem: <name: "configuration" >
+		`,
+			ncFilter:    `<configuration></configuration>`,
+			ncResponse:  errors.New("netconf failure"),
+			wantRetCode: codes.Unknown,
+		}, {
+			desc: "prefxed path",
+			textPrefix: `
+			elem: <name: "configuration" >
+		`,
+			textPbPath: `
+			elem: <name: "version" >
+		`,
+			ncFilter:    `<configuration><version></version></configuration>`,
+			ncResult:    `<configuration><version>ABC</version></configuration>`,
+			wantRetCode: codes.OK,
+			wantRespVal: `ABC`,
+		}, {
+			desc: "ignore nodes not in the schema",
+			textPbPath: `
+			elem: <name: "configuration" >
+		`,
+			ncFilter:    `<configuration></configuration>`,
+			ncResult:    `<configuration><version>ABC</version><notintheschema>XYZ</notintheschema></configuration>`,
+			wantRetCode: codes.OK,
+			wantRespVal: `{
+
+								"version": "ABC"
+
+					}`,
+		}, {
+			desc: "ignore unexpected xml elements in netconf",
+			textPbPath: `
+			elem: <name: "configuration" >
+		`,
+			ncFilter:    `<configuration></configuration>`,
+			ncResult:    `<configuration><version>ABC</version><!-- comment -->></configuration>`,
+			wantRetCode: codes.OK,
+			wantRespVal: `{
+
+								"version": "ABC"
+
+					}`,
+		}}
+	for i := range tests {
+		td := tests[i]
+		t.Run(td.desc, func(t *testing.T) {
+			runTestGet(t, td)
+		})
+	}
+}
+
+// runTestGet requests a path from the server by Get grpc call, and compares if
+// the return code and response value are expected.
+func runTestGet(t *testing.T, tc *getTestCase) {
+
+	mockNc := &mocks.OpSession{}
+	mockNc.On("GetConfigSubtree", tc.ncFilter, ops.RunningCfg, mock.Anything).Return(
+		func(filter interface{}, source string, result interface{}) error {
+			*result.(*string) = tc.ncResult
+			return tc.ncResponse
+		})
+
+	s, err := NewAdapter(model, mockNc)
+	if err != nil {
+		t.Fatalf("error in creating server: %v", err)
+	}
+
+	pbPaths := []*pb.Path{}
+	if !tc.nilPath {
+		pbPath := &pb.Path{}
+		if err := proto.UnmarshalText(tc.textPbPath, pbPath); err != nil {
+			t.Fatalf("error in unmarshaling path: %v", err)
+		}
+		pbPaths = append(pbPaths, pbPath)
+	}
+
+	req := &pb.GetRequest{
+		Path:      pbPaths,
+		Encoding:  pb.Encoding_JSON,
+		UseModels: tc.modelData,
+		Prefix:    getPathPrefix(tc.textPrefix),
+	}
+
+	// Send request
+	resp, err := s.Get(context.TODO(), req)
+
+	// Check return code
+	gotRetStatus, ok := status.FromError(err)
+	if !ok {
+		t.Fatal("got a non-grpc error from grpc call")
+	}
+	if gotRetStatus.Code() != tc.wantRetCode {
```

Comment:
Could use `assert.Equal(t, tc.wantRetCode, gotRetStatus.Code(), "Unexpected return code")` here instead. The values will get printed in the error.

Reply:
Done